### PR TITLE
geth: update to v1.13.15

### PR DESCRIPTION
**Pull request by axelKingsley**
Updates the `go.mod` to use `op-geth` based on upstream version `v1.13.15`

This PR is made against `v1.13.14` branch because there were are required changes to support the a version (.13) which must be built on.

One additional changes required to support the new version beyond changes made in previous updates:
New: 
* use of NewStackTrie changed slightly and has been updated in the single location we use it
Previous:
* `trie` pakcage updated to `triedb` package
* `core` types deprecated for `types` types

One additional change was added which is not strictly required, but enabled:
* Removed the `pin.go` file which fixes a dependency to a specific version, preventing version conflicts. with a more updated geth, we no longer need the pin.

### CODE REVIEW

**Review by trianglesphere — COMMENTED** [review]
LGTM once we update op-geth. FYI I updated the base to op-geth/1.13.14 merge

**Comment by axelKingsley** [comment]
I've changed base back to `develop` because we can take all the changes from the in-between versions by taking this branch, which is a relatively small accumulation. All interior version changes have their own PRs which pass tests and are approved, so merging this one directly is effectively the same as iteratively merging them.

**Review by trianglesphere — APPROVED** [review]

File: op-chain-ops/genesis/genesis.go [hunk 1]
```diff
@@ -200,7 +200,7 @@ func NewL1Genesis(config *DeployConfig) (*core.Genesis, error) {
 		GasUsed:    uint64(config.L1GenesisBlockGasUsed),
 		ParentHash: config.L1GenesisBlockParentHash,
 		BaseFee:    baseFee.ToInt(),
-		Alloc:      map[common.Address]core.GenesisAccount{},
+		Alloc:      map[common.Address]types.Account{},
```
Comment (tynes) [inline op-chain-ops/genesis/genesis.go:203]: If this passes ci then I'm confident in no serialization changes that break things

**Review by tynes — APPROVED** [review]

File: op-chain-ops/genesis/genesis.go (hunk 1 above)
Comment (axelKingsley) [reply op-chain-ops/genesis/genesis.go:203]: Yep, CI on monorepo and op-geth both passing. Per the deprecation note, this appears to have been directly relocated.

**Pull request merged**